src/CachePeer.cc

```diff
@@ -40,7 +41,9 @@ CachePeer::~CachePeer()
     aclDestroyAccessList(&access);
 
 #if USE_CACHE_DIGESTS
-    cbdataReferenceDone(digest);
+    void *digestTmp = nullptr;
+    if (cbdataReferenceValidDone(digest, &digestTmp))
+        peerDigestNotePeerGone(static_cast<PeerDigest *>(digestTmp));
```

Comment:
> >  adjust peerDigestNotePeerGone() to remove the scheduled peerDigestCheck() event (if any).

> This adjustment is not required because  peerDigestNotePeerGone()  does not schedule events.

AFAICT, an adjustment is required to avoid accumulating events scheduled by peerDigestSetCheck() calls. Please see the adjusted PR description for the overall motivation. If there is a problem with that logic, let's discuss.

P.S. That adjustment is probably best placed inside the PeerDigest destructor rather than directly into peerDigestNotePeerGone().

Reply:
Done.